code/game/gamemodes/events/space_weather.dm

```diff
@@ -0,0 +1,296 @@
+
+/datum/event/graveyard/end()
+	command_announcement.Announce("The station wrecks have moved away from the ship.", "Space Graveyard")
+	GLOB.GLOBAL_SANITY_MOD = 1
+
+/datum/storyevent/nebula
+	id = "nebula"
+	name = "Dark matter nebula"
+	weight = 0.8
+	parallel = FALSE
+	event_type = /datum/event/nebula
+	event_pools = list(EVENT_LEVEL_MUNDANE = POOL_THRESHOLD_MUNDANE, EVENT_LEVEL_MODERATE = POOL_THRESHOLD_MODERATE)
+	tags = list(TAG_SCARY, TAG_NEGATIVE)
+
+/datum/event/nebula
+	startWhen = 1
+	announceWhen = 3
+	endWhen = 350
+
+/datum/event/nebula/setup()
+	endWhen = rand(400, 700)
+
+/datum/event/nebula/start()
+	GLOB.GLOBAL_INSIGHT_MOD = 0.5
+
+/datum/event/nebula/announce()
+	command_announcement.Announce("Uncharacteristically high concentrations of dark matter from a nearby nebula currently envelop the ship. Crew might experience certain issues with their mental wellbeing.", "Dark Matter Nebula")
+
+/datum/event/nebula/end()
+	command_announcement.Announce("The dark matter nebula has moved away from the ship.", "Dark Matter Nebula")
+	GLOB.GLOBAL_INSIGHT_MOD = 1
+
+/datum/storyevent/interphase
+	id = "bluespace_interphase"
+	name = "Bluespace Interphase"
+	weight = 0.5
+	parallel = FALSE
+	event_type = /datum/event/interphase
+	event_pools = list(EVENT_LEVEL_MODERATE = POOL_THRESHOLD_MODERATE, EVENT_LEVEL_MAJOR = POOL_THRESHOLD_MAJOR)
+	tags = list(TAG_SCARY, TAG_NEGATIVE)
+
+/datum/event/interphase
+	startWhen = 1
+	announceWhen = 3
+	endWhen = 350
+
+/datum/event/interphase/setup()
+	endWhen = rand(300, 600)
+
+/datum/event/interphase/announce()
+	command_announcement.Announce("The fabric of bluespace has begun to break up, allowing an overlap of parallel universes on different dimensional planes. There is no additional data.", "Bluespace Interphase")
+
+/datum/event/interphase/tick()
+	if(prob(3))
+		var/list/servers = list()
+		for(var/obj/machinery/telecomms/server/S in telecomms_list)
+			if(S.network == "eris" && S.log_entries.len != 0) //yep, only for eris so that non-eris servers don't get involved(duh!)
+				servers += S								//also checks if there are any log entries (not an empty list)
+		if(servers.len != 0)
+			var/obj/machinery/telecomms/server/chosen_server = pick(servers)
+			var/datum/comm_log_entry/C = pick(chosen_server.log_entries)
+			global_announcer.autosay(C.parameters["message"], C.parameters["name"])
+	if(prob(10)) //spooky bluspess ghost
+		var/area/location = random_ship_area()
+		var/mob/living/carbon/human/to_copy = pick(GLOB.human_mob_list)
+		var/mob/ghost = new(location.random_space())
+		ghost.icon = to_copy.icon
+		ghost.icon_state = to_copy.icon_state
+		ghost.dir = to_copy.dir
+		ghost.appearance = to_copy.appearance
+		ghost.mouse_opacity = 0
+		ghost.density = 0
+		sleep(20)
+		if(ghost)//no idea how it could be gone in 20 seconds but gamers will find a way
+			qdel(ghost)
```

Comment:
```suggestion
	if(prob(1))
		var/list/servers = list()
		for(var/obj/machinery/telecomms/server/S in telecomms_list)
			if(S.network == "eris" && LAZYLEN(S.log_entries)) //yep, only for eris so that non-eris servers don't get involved(duh!)
				servers += S								//also checks if there are any log entries (not an empty list)
		if(LAZYLEN(servers))
			var/obj/machinery/telecomms/server/chosen_server = pick(servers)
			var/datum/comm_log_entry/C = pick(chosen_server.log_entries)
			global_announcer.autosay(C.parameters["message"], C.parameters["name"])
	if(prob(5)) //spooky bluspess ghost
		var/area/location = random_ship_area()
		var/mob/living/carbon/human/to_copy = pick(GLOB.human_mob_list)
		var/mob/ghost = new(location.random_space())
		ghost.icon = to_copy.icon
		ghost.icon_state = to_copy.icon_state
		ghost.dir = to_copy.dir
		ghost.appearance = to_copy.appearance
		ghost.mouse_opacity = 0
		ghost.density = 0
		addtimer(CALLBACK(src, .proc/delete_ghost, ghost), 2 SECONDS, TIMER_STOPPABLE)

/datum/event/interphase/proc/delete_ghost(mob/ghost)
	if(!QDELETED(ghost))	//no idea how it could be gone in 20 seconds but gamers will find a way
		qdel(ghost)
```
Less is more